numpy/core/src/umath/simd.inc.src

```diff
@@ -1354,6 +1428,334 @@ sse2_@kind@_BOOL(@type@ * op, @type@ * ip, const npy_intp n)
 /**end repeat**/
 
 #undef VECTOR_SIZE_BYTES
-#endif  /* NPY_HAVE_SSE2_INTRINSICS */
+
+#elif defined NPY_HAVE_VSX_INTRINSICS
+
+/*
+ *****************************************************************************
+ **                           FLOAT LOOPS
+ *****************************************************************************
+ */
+
+// Pack 4x 32-bit vectors into 1x 8-bit vector
+NPY_FINLINE NPY_GCC_TARGET_VSX __vector unsigned char
+pack_vf32_vu8(__vector unsigned int v1, __vector unsigned int v2,
+              __vector unsigned int v3, __vector unsigned int v4)
+{
+    __vector unsigned char mask = vec_splats((unsigned char)0x1);
+    __vector unsigned short p1 = vec_pack(v1, v2);
+    __vector unsigned short p2 = vec_pack(v3, v4);
+    __vector unsigned char r = vec_pack(p1, p2);
+                           r = vec_and(r, mask);
+    return r;
+}
+
+// Pack 8x 64-bit vectors into 1x 8-bit vector
+NPY_FINLINE NPY_GCC_TARGET_VSX __vector unsigned char
+pack_vf64_vu8(__vector unsigned long long v1, __vector unsigned long long v2,
+              __vector unsigned long long v3, __vector unsigned long long v4,
+              __vector unsigned long long v5, __vector unsigned long long v6,
+              __vector unsigned long long v7, __vector unsigned long long v8)
+{
+    __vector unsigned int p1 = vec_pack(v1, v2);
+    __vector unsigned int p2 = vec_pack(v3, v4);
+    __vector unsigned int p3 = vec_pack(v5, v6);
+    __vector unsigned int p4 = vec_pack(v7, v8);
+    return pack_vf32_vu8(p1, p2, p3, p4);
+}
+
+/**begin repeat
+ * #TYPE   = FLOAT, DOUBLE#
+ * #type   = npy_float, npy_double#
+ * #steps  = 4, 2#
+ * #double = 0, 1#
+ * #ivtype = __vector float, __vector double#
+ * #ovtype = __vector unsigned int, __vector unsigned long long#
+ * #otype  = unsigned int, unsigned long long#
+ */
+/**begin repeat1
+ * #kind = equal, not_equal, greater, greater_equal, less, less_equal#
+ * #simd = cmpeq, cmpne, cmpgt, cmpge, cmplt, cmple #
+ * #OP   = ==, !=, >, >=, <, <=#
+ */
+
+/*
+ * Each iteration of the main loop computes 16 elements of the output, which
+ * are packed into an 8-bit vector. In order to generate the 16 elements of the
+ * output, it is necessary to load/compute: 1) 4x 32-bit vectors, for FLOAT; or
+ * 2) 8x 64-bit vectors, for DOUBLE.
+ */
+
+static void NPY_GCC_OPT_3 NPY_INLINE NPY_GCC_TARGET_VSX
```

Comment:
please use universal intrinsics instead, it will require adding new intrinsics for packing boolean values, the rest of intrinsics already implemented.
```
npyv_b8 npyv_pack_b8_b16(npyv_b16 a, npyv_b16 b);
npyv_b8 npyv_pack_b8_b32(npyv_b32 a, npyv_b32 b, npyv_b32  c, npyv_b32 d);
npyv_b8 npyv_pack_b8_b64(npyv_b64 a, npyv_b64 b, npyv_b64  c, npyv_b64 d, npyv_b64 e, npyv_b64 f, npyv_b64  g, npyv_b64 h);
```